x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -142,10 +143,13 @@ public RBACEngine(
     @Override
     public void resolveAuthorizationInfo(RequestInfo requestInfo, ActionListener<AuthorizationInfo> listener) {
         final Authentication authentication = requestInfo.getAuthentication();
-        rolesStore.getRoles(
-            authentication,
-            listener.delegateFailureAndWrap((l, roleTuple) -> l.onResponse(new RBACAuthorizationInfo(roleTuple.v1(), roleTuple.v2())))
-        );
+        rolesStore.getRoles(authentication, listener.delegateFailureAndWrap((l, roleTuple) -> {
+            if (roleTuple.v1() == Role.EMPTY_RESTRICTED_BY_WORKFLOW || roleTuple.v2() == Role.EMPTY_RESTRICTED_BY_WORKFLOW) {
+                l.onFailure(new ElasticsearchAccessRestrictedException("access restricted by workflow"));
+            } else {
+                l.onResponse(new RBACAuthorizationInfo(roleTuple.v1(), roleTuple.v2()));
+            }
+        }));
```

Comment:
It feels a bit weird that the overloading method below
```
resolveAuthorizationInfo(Subject subject, ActionListener<AuthorizationInfo> listener)
```
does nothing for empty-restricted role. 

Looking at its use cases, hasPrivileges, profileHasPrivileges, resolving remote indices for RCS 2.0, I think there might be things that we need dig deeper.

* HasPrivileges - If the code reaches the overloading method, it means the user must have a non-empty-restricted role. So this use case should be fine.
* ProfileHasPrivileges - The API is called by Kibana to check an end-user's privileges. In this case, what should be the workflow for the end-user? It effectively will always behave as no incoming-workflow because ProfileHasPrivileges API does not belong to any workflow. This might not be correct in all cases. But API keys cannot have profiles so it is not an issue today.
* Resolving remote indices - This has two sub-use-cases. 
  * The first one is ad-hoc query and this should be fine because when the code reaches here, we can already be sure the user has a non-empty-restriced role. 
  * The second one is a persistent task which serializes user's authentication object as part of the job definition. Since workflow is another header in threadContext, it should also be serialized for the job. Now this becomes trickly, the serialized workflow would be the API that creates the persistent task, e.g. for transform jobs, this is the PutTransform API. But when the task runs, it runs a search request which would corresponds to a different REST API. In this case, how should workflow work? 

I am not quite sure yet nhow all these could be rationalized into a consistent behaviour. Fortunately, none of them is likely a pratical problem today. So we can revisit it later. 

For now, I suggest we put a TODO comment in the below method and explain briefly why we are leaving it out in this PR.